examples/voice-assistant/llamaindex-rag/chat_engine.py

```diff
@@ -0,0 +1,56 @@
+import os
+
+from dotenv import load_dotenv
+from livekit.agents import AutoSubscribe, JobContext, WorkerOptions, cli, llm
+from livekit.agents.voice_assistant import VoiceAssistant
+from livekit.plugins import deepgram, llama_index, openai, silero
+from llama_index.core import (
+    SimpleDirectoryReader,
+    StorageContext,
+    VectorStoreIndex,
+    load_index_from_storage,
+)
+from llama_index.core.chat_engine.types import ChatMode
+
+load_dotenv()
+
+# check if storage already exists
+PERSIST_DIR = "./chat-engine-storage"
+if not os.path.exists(PERSIST_DIR):
+    # load the documents and create the index
+    documents = SimpleDirectoryReader("data").load_data()
+    index = VectorStoreIndex.from_documents(documents)
+    # store it for later
+    index.storage_context.persist(persist_dir=PERSIST_DIR)
+else:
+    # load the existing index
+    storage_context = StorageContext.from_defaults(persist_dir=PERSIST_DIR)
+    index = load_index_from_storage(storage_context)
+
+chat_engine = index.as_chat_engine(chat_mode=ChatMode.CONTEXT)
```

Comment:
isnt this outside though?

Reply:
it would need to be created in the entrypoint function so its not shared across rooms if my undersranding is correct